.github/workflows/nrp-build.yml

```diff
@@ -19,3 +24,47 @@ jobs:
       artifact: policy-packages
       machine-config: true
       release: true
+
+  release:
+    if: ${{ github.event.inputs.release }}
+    name: Release
+    needs: package
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: policy-packages
+
+      - name: Update json template
+        run: |
+          set -xe
+          asb_artifact=AzureLinuxBaseline.zip
+          asb_hash="$(sha256sum ${asb_artifact} | awk '{print $1}')"
+          asb_uri="https://github.com/${{ github.repository }}/releases/download/${{ github.event.inputs.release }}/${asb_artifact}"
+          asb_template=./src/adapters/mc/asb/AzureLinuxBaseline_DeployIfNotExists.json.tmpl
+          asb_output=./src/adapters/mc/asb/AzureLinuxBaseline_DeployIfNotExists.json
+
+          ssh_artifact=LinuxSshServerSecurityBaseline.zip
+          ssh_hash="$(sha256sum ${ssh_artifact} | awk '{print $1}')"
+          ssh_uri="https://github.com/${{ github.repository }}/releases/download/${{ github.event.inputs.release }}/${ssh_artifact}"
+          ssh_template=./src/adapters/mc/ssh/LinuxSshServerSecurityBaseline_DeployIfNotExists.json.tmpl
+          ssh_output=./src/adapters/mc/ssh/LinuxSshServerSecurityBaseline_DeployIfNotExists.json
+
+          sed -e "s|@HASH@|${asb_hash}|g" -e "s|@URI@|${asb_uri}|g" ${asb_template} > ${asb_output}
+          sed -e "s|@HASH@|${ssh_hash}|g" -e "s|@URI@|${ssh_uri}|g" ${ssh_template} > ${ssh_output}
+
+      - name: Create or update release with new artifacts
```

Comment:
>- name: Create or update release with new artifacts [](http://example.com/codeflow?start=6&length=51)

Is this creating a new release for every run? Who's going to delete that release when done with it?

Please understand these builds have no 1:1 direct relationship to the test policy definitions (which should be changed with this dynamic pattern, they do not point to these internal builds, they point to specific releases for validation, etc).

Reply:
Having said that, I see value in automating updates to test policy definitions that match these GitHub produced policy packages for test for now, and in near future, for the policy packages built in OneBranch and copied here. 

But not to these reference policy definitions, but instead to copies of them that are published under the same GitHub release as the policy packages. So references remain the same, but new dynamic test policy are generated for each set of packages. What do you think of this idea, Jeremi?

Reply:
If you don't provide a release name then no release will be created. If you provide the name of an existing release the artifacts will be updated. Releases can be deleted manually.

Reply:
I imagine we will continue using the "test_policy_package" release name and create new releases around certain milestones (previews). Temporary releases can be cleaned up but they're not going to cause problems if they are kept.

Reply:
> If you don't provide a release name then no release will be created. If you provide the name of an existing release the artifacts will be updated. Releases can be deleted manually.

Thanks for your reply! You mean, provide a release name where exactly? This workflow is automatically invoked for the MC tests. How would this workflow switch between adding a release name and not?

Reply:
> I imagine we will continue using the "test_policy_package" release name and create new releases around certain milestones (previews). Temporary releases can be cleaned up but they're not going to cause problems if they are kept.

Actually, as explained in my other larger comment, we need these policy definitions to remain as-is, manually updated when necessary, and that's not for official releases (for those the policy definitions are checked in modified to a different repository) but for internal validation and self-hosting, including my own.

Do you wish to propose a faster way of selfhosting, like a 'canary' or 'insiders-fast' ring with this? I am not opposed to having two rings for internal self-hosting of the policies, but we need to keep the existing 'ring' as-is.

If that's what you intend with this PR, what would be the reasons behind adding such a 'canary' self-hosting option? Let's discuss more, this sounds like a very interesting proposal, please help me to understand more about what you are trying to achieve.